core/src/main/java/io/grpc/ChannelImpl.java

```diff
@@ -283,25 +287,35 @@ public void start(Listener<RespT> observer, Metadata.Headers headers) {
       try {
         transport = obtainActiveTransport();
       } catch (RuntimeException ex) {
-        stream = new NoopClientStream();
-        listener.closed(Status.INTERNAL.withDescription("Failed starting transport").withCause(ex),
-            new Metadata.Trailers());
+        closeCallPrematurely(listener,
```

Comment:
I purposefully only set stream once (and late) so that if we accidentally started using the listener too early we would get a NullPointerException.